CMakeLists.txt

```diff
@@ -43,6 +43,35 @@ if (NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE Release CACHE STRING "choose 'Release' as default build type" FORCE)
 endif ()
 
+if(NOT CMAKE_CXX_COMPILER)
+    set(CMAKE_CXX_COMPILER g++-7)
+endif()
+
+# Fill backends with inference software directory
+if(DEFINED TENSORRT_DIR)
```

Comment:
I found those conditions are bringing drawbacks for users who export path setter to env to set them in IDE, like Clion.
They can disable/enable backend just by changing `MMDEPLOY_TARGET_BACKENDS` without changing path setter.
![image](https://user-images.githubusercontent.com/4560679/184829472-d43c7d89-3500-4c79-9e4c-2762c4dad4fb.png)
But now, besides updating `MMDEPLOY_TARGET_BACKENDS`, they have also to delete or add the path setter

Actually, by exporting path setter to env, users don't need to pass them to make at all.
If there are multi versions, they have the ability to make up scripts and source them according to different version.
-